### ISSUE TITLE
NPM package changed directory format, making NPM fail (since 2.14.7)

### ISSUE THREAD

**Issue by paulrutter** [eirslett/frontend-maven-plugin#328]
I've upgraded NPM from 2.7.4 to 2.14.7, and NPM does not work any more since.
Something about "npm-cli" not being where its supposed to be.

I've been able to debug this and seen that the newer NPM tgz files have a different format.
Where the old ones always had a "package" directory, the newer ones do not have the "package" directory, but a "npm-2.14.7" directory instead.

Please fix this in the maven plugin.
As a workaround i've hosted NPM myself, with a modified NPM tgz file containing a "package" directory.

**Comment by antoinebrault** [comment]
+1

**Comment by cameroncan** [comment]
Can we get this fixed? The pull request fixes my issue where the plugin looks for npm/bin/npm-cli.js, when it is installed to npm-3.8.3/bin/npm-cli.js

**Comment by eirslett** [comment]
I can't reproduce this problem... there's an integration test in frontend-maven-plugin that uses npm 3. Shouldn't it fail?

**Comment by cameroncan** [comment]
The issue seems to be when I specify the node and npm install paths through <nodeDownloadRoot> and <npmDownloadRoot> (to our internal nexus server). When I specify only the versions as the tests do, this works just fine for me. I am going this route for now and will dig in deeper later to get it working when pulling from our nexus server.

**Comment by paulrutter** [comment]
Yes, the issue only occurs when using the <npmDownloadRoot> tag, i forgot to mention this!

**Comment by eirslett** [comment]
Is there a difference in the checksums of the file that is downloaded from the official repo and the one from internal nexus?

**Comment by art-at-sky** [comment]
I'm using `nodeDownloadRoot` and `npmDownloadRoot` too, which is I why I encountered this issue and submitted a PR

**Comment by paulrutter** [comment]
Any update on when/if the PR will be merged?

**Comment by eirslett** [comment]
I'm closing this issue now, because it's very old. I think it was fixed a while ago.
